src/main/webapp/app/shared/consistency-check/consistency-check.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { ConsistencyCheckService } from 'app/shared/consistency-check/consistency-check.service';
+import { JhiAlertService } from 'ng-jhipster';
+import { ConsistencyCheckError } from 'app/entities/consistency-check-result.model';
+import { getCourseId } from 'app/entities/exercise.model';
+
+@Component({
+    selector: 'jhi-consistency-check',
+    templateUrl: './consistency-check.component.html',
+})
+export class ConsistencyCheckComponent implements OnInit {
+    @Input() id: number;
+    @Input() checkType: CheckType;
+
+    errors: ConsistencyCheckError[];
+    courseId: number | undefined;
+
+    constructor(private activeModal: NgbActiveModal, private consistencyCheckService: ConsistencyCheckService, private jhiAlertService: JhiAlertService) {}
+
+    ngOnInit(): void {
+        if (this.checkType === CheckType.PROGRAMMING_EXERCISE) {
+            this.consistencyCheckService.checkConsistencyForProgrammingExercise(this.id).subscribe(
+                (errors) => {
```

Comment:
Having the successful case named `errors` and the failure one named `err` is quite hard to read at a glance. A different name like e.g. `inconsistencies` (maybe also for the attribute of the class?) would probably help.

Reply:
I agree.. Renamed to `inconsistencies` in last commit :+1: